### DISCUSSION TITLE
Is it possible to display a COG's default palette?

### DISCUSSION THREAD

**Discussion by jmarokhovsky** [opengeos/leafmap#858] [Q&A] [answered]
Currently in **leafmap 0.36.2** not adding a palette to the method call results in the palette being black.
```python
        self.add_cog_layer(
            aqua_url,
            "Honduras Landcover 2022",
            opacity=0.5,
        )
```
<img width="267" alt="Screenshot 2024-07-31 at 12 04 52" src="https://github.com/user-attachments/assets/c0cd639c-d6ea-4ecb-847c-42d376d49b8b">

If I specify a palette those colors do display:
```python
        self.add_cog_layer(
            aqua_url,
            "Honduras Landcover 2022",
            opacity=0.5,
            palette="viridis",
        )
```
<img width="261" alt="Screenshot 2024-07-31 at 12 05 37" src="https://github.com/user-attachments/assets/47fa78aa-ae1a-4237-883b-0808430d661e">

I do know the palette exists since this is the same COG loaded into QGIS:
<img width="683" alt="Screenshot 2024-07-31 at 12 08 47" src="https://github.com/user-attachments/assets/e326753c-2308-4c23-82f8-68cbc4327b70">
<img width="1474" alt="Screenshot 2024-07-31 at 12 09 17" src="https://github.com/user-attachments/assets/bbf5e76f-a0d9-43ee-ac97-00390d5531ed">

Is this a bug or is there a word that I'm not finding in the docs?

**Related question:** Is there a reason to use the `colormap` argument instead of the `palette` one? I only see reference to `colormap` in the API docs.

**Comment by giswqs** [comment, marked as answer]
The `add_cog_layer` function relies on TiTiler. You can see the available parameters [here](https://developmentseed.org/titiler/endpoints/cog/#tiles). I am not sure if TiTiler can render the default palette from the tile. Usually, you have to specify either `colormap` or `colormap_name`.
